projects/storefrontlib/src/lib/suffix-routes/suffix-url-matcher.ts

```diff
@@ -0,0 +1,44 @@
+import { UrlSegment, UrlMatchResult } from '@angular/router';
+
+export function findLastIndex<T>(elements: T[], predicate: (el: T) => boolean) {
+  for (let index = elements.length - 1; index >= 0; index--) {
+    if (predicate(elements[index])) {
+      return index;
+    }
+  }
+  return -1;
+}
+
+/**
+ * Matches the pattern '[ ** / ] marker / :paramName [ / ** ]'
+ *
+ * @param marker phrase that indicates the start of the match
+ * @param paramName name of the parameter present after the marker
+ * @param precedingParamName name of the parameter for every preceding url segment
+ *        i.e. `urlSegment` will result in `urlSegment0`, `urlSegment1`, ...
+ */
+export function suffixUrlMatcher(
+  marker: string,
+  paramName: string,
+  precedingParamName: string = 'urlSegment'
```

Comment:
what about changing this default to less verbose, like `'segment'` or even `''`?